pkg/controllers/managementuser/healthsyncer/healthsyncer.go

```diff
@@ -82,20 +86,29 @@ func (h *HealthSyncer) getComponentStatus(cluster *v3.Cluster) error {
 	// As of k8s v1.14, kubeapi returns a successful ComponentStatuses response even if etcd is not available.
 	// To work around this, now we try to get a namespace from the API, even if not found, it means the API is up.
 	if _, err := h.k8s.CoreV1().Namespaces().Get(ctx, "kube-system", metav1.GetOptions{}); err != nil && !apierrors.IsNotFound(err) {
-		return condition.Error("ComponentStatsFetchingFailure", errors.Wrap(err, "Failed to communicate with API server during namespace check"))
+		return condition.Error("ComponentStatusFetchingFailure", errors.Wrap(err, "Failed to communicate with API server during namespace check"))
 	}
 
 	cluster.Status.ComponentStatuses = []v32.ClusterComponentStatus{}
-	k8sVersion, err := semver.Parse(fmt.Sprintf("%s.%s.0", cluster.Status.Version.Major, cluster.Status.Version.Minor))
+	if cluster.Status.Version == nil {
+		return nil
+	}
+	parts := versionMatchRE.FindStringSubmatch(cluster.Status.Version.String())
+	if parts == nil {
+		return condition.Error("ComponentStatusFetchingFailure", fmt.Errorf("Failed to parse cluster status version %s",
+			cluster.Status.Version.String()))
+	}
+	k8sVersion, err := semver.Parse(parts[1])
```

Comment:
is there any chance of parts having a length of 1?  97 could be `parts == nil || len(parts) < 2`

Reply:
I had a similar question, but they seem to be accessing indices directly, 
https://github.com/kubernetes/apimachinery/blob/v0.22.3/pkg/util/version/version.go#L47

```
	if parts == nil {
		return nil, fmt.Errorf("could not parse %q as version", str)
	}
	numbers, extra := parts[1], parts[2]
```
I'll add a check to be safe.